### ISSUE TITLE
Change 2 scripts

### ISSUE THREAD

**Issue by HongThatCong** [mandiant/flare-floss#939]
In my opinion, I think it is necessary to change/addition the following 2 scripts, or maybe merge those 2 scripts into 1.

1. render-ida-import-script.py:
Runs as an IDAPython script, allowing users to select floss json result file to apply stacks/functions comments directly into IDA.
No need to generate an intermediate script.

2. idaplugin.py
Added parse language for Go/Rust files

3. Combine the above two scripts into a new IDAPython script, running entirely in IDA

Tks and brgds
HTC

**Comment by mr-tz** [comment]
Thanks for the suggestions. There's definitely room for improvement and potential to combine the scripts.

**Comment by Ananthavijay** [comment]
Can I work on this issue?

**Comment by williballenthin** [comment]
go for it @Ananthavijay

**Comment by Atlas-64** [comment]
hi @HongThatCong  @williballenthin , if it hasn't been assigned yet I would love taking a hack at it

**Comment by mr-tz** [comment]
Go for it 👍

**Comment by Atlas-64** [comment]
Hi @mr-tz sorry for my confusion but I think eventually I would have to have access to IDA Pro to run the prewritten tests after making changes to these plugins(i.e making them one ). Asking cuz I am trying to currently start working on a personal IDA plython plugin and hit this wall. And on searching found that the free version doesn't let me interact with plugins or build IDA Python. Correct me if I am wrong, thanks for everyone's time .

**Comment by mr-tz** [comment]
Yes, correct, you need a non-free version of IDA Pro to run IDAPython scripts.

**Comment by HongThatCong** [comment]
Someone build IDAPython for IDA Free 8.3. 
I found this link on ExeTools forum.
@Atlas-64 you can try with it.
Uses idapyswitch to correct it with your Python version.

Or you can up the source file to here, I can test.

**Comment by Atlas-64** [comment]
thanks a lot @HongThatCong  I will check the build out if I face issues or I end up needing help on testing I will reply here . Thanks
for the help.
Also if you still have the ExeTools post associated with the free IDAPython build do link it here if possible . Thanks again :D .

**Comment by HongThatCong** [comment]
<https://forum.exetools.com/showpost.php?p=127922&postcount=14>

**Comment by rainisveryswag** [comment]
Hi, I’m applying for GSoC 2025 and would love to work on this as my small contribution. Can I take it?

**Comment by codesensei-tushar** [comment]
Hi @mr-tz, checking in on this. If no one is currently working on this, I can take it over.

**Comment by williballenthin** [comment]
do you have an IDA license @codesensei-tushar ?

**Comment by codesensei-tushar** [comment]
I don't have a license @williballenthin but I'd like to give a shot at this using the IDAPython for IDA Free 8.3 method mentioned by @HongThatCong 
> Someone build IDAPython for IDA Free 8.3. 
> 
> I found this link on ExeTools forum.
> 
> @Atlas-64 you can try with it.
> 
> Uses idapyswitch to correct it with your Python version.
> 
> 
> 
> https://www.mediafire.com/file/8xeuzp14dhov4vd/idapython83.zip/file
> 
> 
> 
> Or you can up the source file to here, I can test. 

I'll create a PR using that and we can see if it passes the necessary checks. If that method fails, we can figure out another way.

**Comment by vee1e** [comment]
@mr-tz @HongThatCong can you give a brief description of the expected functionality of the new script:

1. should it support two modes, JSON import and Live analysis, or just one?
2. what should happen to the old files, should they be marked deprecated or removed entirely
3. should it interface with IDA as a plugin or an IDAPython script? If the latter, I think the idaplugin.py script should be kept and render-ida-import-script.py can contain the new changes.
